modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/internal/upgrade/v14_0_0/test/ObjectDefinitionUpgradeProcessTest.java

```diff
@@ -0,0 +1,456 @@
+
+	@Before
+	public void setUp() throws Exception {
+		_originalSystemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager(
+					CPDefinition.class.getSimpleName());
+
+		SystemObjectDefinitionManager newSystemObjectDefinitionManager =
+			(SystemObjectDefinitionManager)ProxyUtil.newProxyInstance(
+				SystemObjectDefinitionManager.class.getClassLoader(),
+				new Class<?>[] {SystemObjectDefinitionManager.class},
+				(proxy, method, args) -> {
+					if (Objects.equals(
+							method.getName(), "getPrimaryKeyColumn")) {
+
+						return CPDefinitionTable.INSTANCE.CPDefinitionId;
+					}
+
+					if (Objects.equals(method.getName(), "getObjectFields")) {
+						List<ObjectField> objectFields =
+							(List<ObjectField>)method.invoke(
+								_originalSystemObjectDefinitionManager, args);
+
+						objectFields.set(
+							4,
+							new TextObjectFieldBuilder(
+							).dbColumnName(
+								"CPDefinitionId"
+							).labelMap(
+								LocalizedMapUtil.getLocalizedMap(
+									RandomTestUtil.randomString())
+							).name(
+								"productId"
+							).system(
+								true
+							).build());
+
+						return objectFields;
+					}
+
+					return method.invoke(
+						_originalSystemObjectDefinitionManager, args);
+				});
+
+		_serviceTrackerMap = ReflectionTestUtil.getFieldValue(
+			_systemObjectDefinitionManagerRegistry, "_serviceTrackerMap");
+
+		ReflectionTestUtil.setFieldValue(
+			_systemObjectDefinitionManagerRegistry, "_serviceTrackerMap",
+			ProxyUtil.newProxyInstance(
+				ServiceTrackerMap.class.getClassLoader(),
+				new Class<?>[] {ServiceTrackerMap.class},
+				(proxy, method, args) -> {
+					if (Objects.equals(method.getName(), "getService") &&
+						Objects.equals(args[0], "CPDefinition")) {
+
+						return newSystemObjectDefinitionManager;
+					}
+
+					return method.invoke(_serviceTrackerMap, args);
+				}));
+
+		_originalObjectDefinitionLocalService =
+			(ObjectDefinitionLocalService)
+				ReflectionTestUtil.getAndSetFieldValue(
+					_portalInstanceLifecycleListener,
+					"_objectDefinitionLocalService",
+					ProxyUtil.newProxyInstance(
+						ObjectDefinitionLocalService.class.getClassLoader(),
+						new Class<?>[] {ObjectDefinitionLocalService.class},
+						(proxy, method, args) -> {
+							if (!Objects.equals(
+									method.getName(),
+									"addOrUpdateSystemObjectDefinition")) {
+
+								return method.invoke(
+									_objectDefinitionLocalService, args);
+							}
+
+							SystemObjectDefinitionManager
+								systemObjectDefinitionManager =
+									(SystemObjectDefinitionManager)args[2];
+
+							if (!Objects.equals(
+									systemObjectDefinitionManager.getName(),
+									"CPDefinition")) {
+
+								return method.invoke(
+									_objectDefinitionLocalService, args);
+							}
+
+							args[2] = newSystemObjectDefinitionManager;
+
+							return method.invoke(
+								_objectDefinitionLocalService, args);
+						}));
+
```

Comment:
@pedro-oliveira446 for better reading, I would separate this code by moving it to the `setUpClass` method